LongLookupTables/make_lookup_tables.py

```diff
@@ -0,0 +1,389 @@
+# -*- coding: utf-8 -*-
+
+"""
+Script to generate of the lookup tables problem.
+
+Running this script will save the following files in /dir/ or /dir/sample<i>/ if n_samples > 1:
+- train.tsv
+- validation.tsv
+- heldout_inputs.tsv
+- heldout_compositions
+- heldout_tables
+- new_compositions
+- longer_seen_1.tsv
+- longer_incremental_1.tsv
+- longer_new_1.tsv
+...
+- longer_seen_n.tsv
+- longer_incremental_n.tsv
+- longer_new_n.tsv
+
+with n is the max number of additional compositions in test compared to train.
+
+Help : `python make_lookup_tables.py -h`
+"""
+
+from __future__ import unicode_literals, division, absolute_import, print_function
+
+import sys
+import os
+import itertools
+import random
+import operator
+import warnings
+import argparse
+from functools import reduce
+
+import numpy as np
+import pandas as pd
+from sklearn.model_selection import train_test_split
+
+
+### MAIN ###
+def parse_arguments(args):
+    parser = argparse.ArgumentParser(description="Script to generate of the lookup tables problem.",
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument('-d', '--dir', default='.', help='Path to the directory where to save the generated data.')
+    parser.add_argument('-s', '--n-samples', type=int, default=1, help='Number of different samples to generate. If greater than 1, will save the files for a single sample in /dir/sample<i>/*.tsv.')
+
+    parser.add_argument('-v', '--validation-size', type=float, default=0.1, help='Percentage of training set to use as validation.')
+    parser.add_argument('-c', '--max-composition-train', type=int, default=4, help='Max length of compositions in training set.')
+    parser.add_argument('-t', '--n-unary-tables', type=int, default=8, help='Number of different lookup tables.')
+    parser.add_argument('-T', '--n-heldout-tables', type=int, default=2, help='Number of tables that would only be seen in unary.')
+    parser.add_argument('-C', '--n-heldout-compositions', type=int, default=50, help='Number of compositions to randomly remove.')
+    parser.add_argument('-I', '--n-heldout-inputs', type=int, default=2, help='Number of inputs to heldout from training.')
+    parser.add_argument('-l', '--n-longer', type=int, default=2, help='Number of additional tables to add to `longer` test data.')
```

Comment:
This one I don't understand. Does this concern the length of longer?

Reply:
Yes the number of rows in `longer_*.tsv` is simply bounded as it could get huge really quickly : nearly 10 billion for 8 different tables and 8 inputs and 10 compositions.

Reply:
ah OK I see. Perhaps just change the help to something like "length of longer compositions" or something like that, at first I interpreted this as if new tables should be added (e.g. t25)